src/core/thread/mle.cpp

```diff
@@ -259,7 +263,17 @@ void Mle::Stop(StopMode aMode)
     SetRole(kRoleDisabled);
 
 exit:
-    return;
+    if (mDetachGracefullyCallback != nullptr)
```

Comment:
Should we also stop `mDetachGracefullyTimer` even if the callback is nullptr?

Reply:
Done.